File: Composite Pattern/LegoBauteil_java/src/main/Lego/LegoPiece.java
package main.Lego;

public final class LegoPiece implements ILego {
    private double price;
    private int size;
    private String color;

    public LegoPiece() {
        this.setColor("");
        this.setPrice(0);
        this.setSize(0);
    }

    public LegoPiece(final String color, final double price, final int size) {
        this.setColor(color);
        this.setPrice(price);
        this.setSize(size);
    }

    @Override
    public double getPrice() {
        return price;
    }

    public void setPrice(final double price) {
        this.price = price;
    }

    public int getSize() {
        return size;
    }

    public void setSize(final int size) {
        this.size = size;
    }

    public String getColor() {
        return color;
    }

    public void setColor(final String color) {
        this.color = color;
    }
}

File: Composite Pattern/LegoBauteil_cpp/LegoPiece.cpp
#include "LegoPiece.h"

LegoPiece::LegoPiece()
{
	this->setColor("");
	this->setPrice(0);
	this->setSize(0);
}

LegoPiece::LegoPiece(const std::string& color, const double& price, const int& size)
{
	this->setColor(color);
	this->setPrice(price);
	this->setSize(size);
}

auto LegoPiece::getPrice() const -> double
{
	return this->_price;
}

auto LegoPiece::getColor() const -> std::string
{
	return this->_color;
}

auto LegoPiece::getSize() const -> int
{
	return this->_size;
}

auto LegoPiece::setPrice(const double& price) -> void
{
	this->_price = price;
}

auto LegoPiece::setColor(const std::string& color) -> void
{
	this->_color = color;
}

auto LegoPiece::setSize(const int& size) -> void
{
	this->_size = size;
}
